File: spring-boot-api/src/main/java/dev/codewithsam/springbootapi/health/HealthController.java
package dev.codewithsam.springbootapi.health;

import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * The Spring Boot half of the Episode 0 baseline.
 *
 * Deliberately hand written rather than pulled from Actuator, because the point
 * of the episode is to put the SAME conceptual endpoint beside the NestJS one
 * and let the viewer compare the two stacks answering it. Actuator arrives
 * later, once there is a reason for it.
 */
@RestController
public class HealthController {

  @GetMapping("/health")
  public Map<String, String> check() {
    return Map.of("status", "UP");
  }
}

File: nestjs-api/src/health/health.controller.ts
import { Controller, Get } from '@nestjs/common';

/**
 * The one endpoint Episode 0 is graded on.
 *
 * It exists in BOTH applications so the viewer can see the same conceptual
 * thing answered by two stacks: Spring Boot on 8080, NestJS on 3000. Nothing
 * else in the repository matters until both of these return 200.
 */
@Controller('health')
export class HealthController {
  @Get()
  check(): { status: string } {
    return { status: 'UP' };
  }
}
